script.js

```diff
@@ -1,4 +1,84 @@
+var winCount = 0;
+var numDraws = 0;
+var lossCount = 0;
+var currentMode = "waiting for user name";
+var userName = "";
+var myOutputValue = "";
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (currentMode == "waiting for user name") {
+    // set the name of user
+    userName = input;
+
+    currentMode = "game mode";
+    myOutputValue = `Your name is ${userName}.`;
+    return myOutputValue;
+  } else if (currentMode == "game mode") {
+    var spsGenerator = spsOptions();
+    console.log(spsGenerator);
+    var option = "";
+
+    //if computer rolls 1, it will show paper
+    if (spsGenerator == 1) {
+      option = "paper";
+    }
+    //if computer rolls 2, it will show scissors
+    if (spsGenerator == 2) {
+      option = "scissors";
+    }
+    //if computer rolls 3, it will show stone
+    if (spsGenerator == 3) {
+      option = "stone";
+    }
+
+    console.log(option);
+
+    // DRAW
+    if (input == option && currentMode) {
+      var myOutputValue = `It's a draw ${userName}! <br><br> You rolled ${input} and the computer rolled ${option}.`;
```

Comment:
`myOutputValue` was already declared earlier in line 6, so there's no need to use `var` again here, although it might seem silly that you can use it and it works, but generally bad practice.